### PR TITLE
fix(gatsby-source-drupal): delete relationships to now deleted nodes

### PR DESCRIPTION
**Pull request by KyleAMathews**
We weren't cleaning these up which would cause schema/query errors

### CODE REVIEW

File: packages/gatsby-source-drupal/src/utils.js [hunk 1]
```diff
@@ -110,6 +110,68 @@ const handleReferences = (
...
+              node.relationships[key] = value
+            }
+          }
+        })
+
+        // Remove deleted node from array of referencedNodes
+        referencedNodes = referencedNodes.filter(nId => nId !== deletedNode.id)
+        referencedNodesLookup.set(node, referencedNodes)
+      }
+    })
```
Comment (wardpeet) [inline packages/gatsby-source-drupal/src/utils.js:167]: I'm a bit lost here, shouldn't we also call createNode to re-create the nodes and deleteNode? Now we're mutating node.relationships without saving it, wouldn't it cause problems?

**Review by wardpeet — COMMENTED** [review]
I asked some extra questions to get more info as I'm not super familiar with Drupal source plugin

File: packages/gatsby-source-drupal/src/utils.js (hunk 1 above)
Comment (KyleAMathews) [reply packages/gatsby-source-drupal/src/utils.js:167]: Oh right, yeah forgot that part 🙃

File: packages/gatsby-source-drupal/src/utils.js [hunk 2]
```diff
@@ -110,6 +110,68 @@ const handleReferences = (
...
+        node.attributes?.drupal_internal__revision_id,
+        entityReferenceRevisions
+      )
+    )
+  )
+
+  // Remove the deleted node from backRefsNamesLookup
+  backRefsNamesLookup.delete(deletedNode)
+
+  // Remove relationships from other nodes and re-create them.
```
[the remark this answers is not in the document]
Comment (KyleAMathews) [reply packages/gatsby-source-drupal/src/utils.js:135]: Relationships are just `___NODE` pointers atm so we need to manually remove them.

**Comment by KyleAMathews** [comment]
Validated fix on a test Drupal site, things should be ready to go 👍

**Review by smthomas — APPROVED** [review]
This looks good to me! 🚢  it

**Pull request merged**